vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -252,25 +252,44 @@ def _get_unfiltered_data(
 def _get_modified_page_figures(
     ctds_filter: list[CallbackTriggerDict],
     ctds_filter_interaction: list[dict[str, CallbackTriggerDict]],
-    ctds_parameters: list[CallbackTriggerDict],
+    ctds_parameter: list[CallbackTriggerDict],
     targets: list[ModelID],
 ) -> dict[ModelID, Any]:
+    from vizro.models import Filter
+
     outputs: dict[ModelID, Any] = {}
 
+    control_targets = []
+    figure_targets = []
+    for target in targets:
+        if isinstance(model_manager[target], Filter):
+            control_targets.append(target)
+        else:
+            figure_targets.append(target)
+
+    # TODO-NEXT: Add fetching unfiltered data for the Filter.targets as well, once dynamic filters become "targetable"
```

Comment:
Here's the use case where this will not work. Let's say you have a filter (F), a parameter (P) and two graphs (G1, G2). 

Now, let's define targeting "scheme". Right components are targeted by left components. Also, here's a new case where the filter F is also explicitly targeted by the parameter P.  
F -> G1, G2
P -> F, G2.

When on_page_load is triggered: targets are: F, G1, G2, and this function will return correct objects. It will fetch the data_frames for G1 and G2 and calculate new G1, G2 + it will calculate filter F options based on the retrieved data_frames that are fetched for the G1, G2. 

Now, if the parameter action P is triggered: targets are: F, G2, and this function will fetch the data_frame for G2 and calculate new G2 component correctly. However, I can't calculate filter F options because the data_frame for G1 is missing.

So, this comment says that this function has to fetch:
 - data_frames for the targeted graphs (figures actually), (this already works)
 - and also data_frames for the graphs/figures that are targeted by dynamic filters that are targeted by this action. (this has to be added)

In that case, the mentioned fictive parameter action would fetch data_frame for G2, but also the data_frame for the G1 because it's targeted by the filter F. The code that fetches dynamic filter targets data_frames is implemented [here](https://github.com/mckinsey/vizro/commit/161a1e9e34c6c6032f9ff3c3081ec35ffdb262a1), but it's removed until we support that dynamic filters could be targeted from any action (except on_page_load).

Reply:
THanks a lot, that really really helped - we might want to remember this comment haha, or even put it in the code!